app/models/assessment.rb

```diff
@@ -338,10 +341,22 @@ def default_total_score
     problems.sum :max_score
   end
 
+  def update_course_grade_watchlist_instances_if_past_end_at
+    if Time.now >= self.end_at
+      self.course.update_course_grade_watchlist_instances
+    end
+  end
+
 private
 
+  def saved_change_to_grade_related_fields?
+    return (saved_change_to_due_at? or saved_change_to_late_penalty_id? or
+            saved_change_to_max_grace_days? or saved_change_to_version_threshold? or
+            saved_change_to_version_penalty_id?)
+  end
```

Comment:
I think it would be possible to do the time inside the saved_change_to_grade_related_fields instead of having an additional wrapper function

Reply:
Let me know if this is possible?

Reply:
Do you want me to add back the saved_change_to_penalty_id?'s? Sure I can do that! I'll take a note and add it in a separate PR.